src/go/log/log_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
```

Comment:
Ditto best practice here. Allows asserts in tests to continue test execution / summary output? I think maybe? Generally explanation is helpful somewhere in best practices maybe? @hcgatewood

Reply:
ah, `assert` is a misnomer here.

the `testify/assert` library is a reflection/syntactic sugar library to help turn this:

```
if  got != want {
    t.Errorf("X(%s) = %s, want=%s", input, got, want)
}
```

into this:

```
assert.Equal(t, want, got, "X(%s)", input)
```

The first construction uses the compiler to type-check `got` and `want`, but the latter is one line.

Years ago, I used to be a stickler for the first construction ("if golang authors haven't added it to the lang/stdlib, then we don't need it either!"), but the overwhelming majority of folks prefer the latter and it's just for tests so I've relented.

To be fair, a huge block of `if X != Y { ... }` is much harder on the eyes than a set of `assert.Equal(...)` one-liners.

Reply:
Assert use is discussed also in [golang github repo](https://github.com/golang/go/issues/34384) - they don't seem to like it but that seems to align with what you say here about the `Go way of doing things the long way` - I have no horse in this "race" =D Just thought I would note.